### PR TITLE
Add missed packages to packages.builds for 2.1.5

### PR DESCRIPTION
**Pull request by wtgodbe**
I went through all changes to `corefx\src` in release/2.1 since 2.1.0, and I believe that these are all of the packages that should have been shipped for servicing, but weren't.

- System.IO.Pipelines - https://github.com/dotnet/corefx/commit/141c2f4ce32c77b1d83bffc2bd819458dfa12891#diff-25d902c24283ab8cfbac54dfa101ad31
- System.Security.Cryptography.Pkcs - https://github.com/dotnet/corefx/commit/767d4aab5aaa43bd06fbd7490bda8ef05973f3b0#diff-25d902c24283ab8cfbac54dfa101ad31 (among others)
- System.Net.WebSockets.WebSocketProtocol - https://github.com/dotnet/corefx/commit/c6f4eb6cc7ea7f7d1ff629555969f44425de342e#diff-25d902c24283ab8cfbac54dfa101ad31 (note that we did ship this package as part of 2.1.2, but it was removed from `packages.builds` before we could ship the changes from the linked PR. See https://github.com/dotnet/corefx/commit/8dabbd8adc40cec5b12b2d2168fd4ef3e9f08a62#diff-25d902c24283ab8cfbac54dfa101ad31)
- System.Security.Principal.Windows - https://github.com/dotnet/corefx/commit/6ac8b2aabaef2bbdb6b962110209f261017e0fbb#diff-25d902c24283ab8cfbac54dfa101ad31
- System.Net.Http.WinHttpHandler - https://github.com/dotnet/corefx/commit/be49d38171de4c5cb45c4a5e71e9156f05e67404#diff-25d902c24283ab8cfbac54dfa101ad31
- System.Diagnostics.DiagnosticSource - https://github.com/dotnet/corefx/commit/35a009a17866ce41a95f25780503b82183b5ec9e#diff-25d902c24283ab8cfbac54dfa101ad31
- System.Runtime.CompilerServices.Unsafe - https://github.com/dotnet/corefx/commit/5b16645d8a49a65a531f805e1e29c15f6b092b39#diff-25d902c24283ab8cfbac54dfa101ad31 (was also shipped as part of 2.1.2, but then had subsequent changes made that didn't get shipped)

@weshaggard @safern PTAL

CC @danmosemsft 

[edit] Here's the list of the original 2.1 fix PR's
- Fix ReadAsync not returning when cancellation token fires during FlushAsync #31871
- Fix compatibility issues vs NetFX in SignedCms and EnvelopedCms #30432
- Fix WebSocket server split header parsing with large payload #30407
- Duplicate the access token passed to WindowsIdentity.RunImpersonated #30379
- Fix WinHttpHandler when using authenticating proxies #30212
- Add support and tests for HTTP 308 Permanent Redirect #30954
- Adding special target that will get a resource file and pass it to ilasm for S.R.CS.Unsafe so that it has the right version headers #31806 

There are likely other issues that would not be shipped without this PR. The above is just the first change in the library. Eg., https://github.com/dotnet/corefx/pull/31680

### CODE REVIEW

File: src/System.Security.Cryptography.Pkcs/dir.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.0.3.0</AssemblyVersion>
+    <AssemblyVersion>4.0.3.1</AssemblyVersion>
```
[the remark this answers is not in the document]
Comment (bartonjs) [reply src/System.Security.Cryptography.Pkcs/dir.props]: Presumably we should just close #32111 and take this as the "fix everything" change?

[the remark this answers is not in the document]
Comment (weshaggard) [reply src/System.Security.Cryptography.Pkcs/dir.props]: I suspect so.

[the remark this answers is not in the document]
Comment (wtgodbe) [reply src/System.Security.Cryptography.Pkcs/dir.props]: @weshaggard do I also need to update packageIndex.json with the new `AssemblyVersion`s, like https://github.com/dotnet/corefx/pull/32111/files#diff-122916076db7087dbc454352fada61eeR4293? If so I should update the doc as well.

File: src/System.IO.Pipelines/dir.props [hunk 2]
```diff
@@ -2,7 +2,8 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.0.0.0</AssemblyVersion>
+    <AssemblyVersion>4.0.0.1</AssemblyVersion>
```
Comment (weshaggard) [inline src/System.IO.Pipelines/dir.props]: @natemcmaster FYI we are bumping the assembly version of System.IO.Pipelines for 2.1.5 servicing.

File: src/System.Security.Cryptography.Pkcs/dir.props (hunk 1 above)
[the remark this answers is not in the document]
Comment (weshaggard) [reply src/System.Security.Cryptography.Pkcs/dir.props]: I suspect you might. You will hopefully figure out when the all configuration leg fails.

**Comment by safern** [comment]
Will per: https://github.com/dotnet/corefx/pull/32108#issuecomment-420347362 I added commit 4e4dc12 to fix the System.Drawing.Common assembly version.

**Comment by jamshedd** [comment]
Approved for 2.1.5.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 3]
```diff
@@ -3005,7 +3008,8 @@
       ],
       "InboxOn": {},
       "AssemblyVersionInPackageVersion": {
-        "4.0.0.0": "4.5.0"
+        "4.0.0.0": "4.5.1",
```
Comment (ViktorHofer) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: The assembly version's package version should be the first where the assembly was shipped. In this case 4.5.0 and not 4.5.1. Same for other below.

Comment (weshaggard) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: No it should be the latest package that shipped stable. We want to ensure that folks are getting the latest version of any changes with the same assembly version.

Comment (ViktorHofer) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: hmm makes sense but @ericstj told me the opposite in a random PR which I can't find right now.

**Comment by weshaggard** [comment]
The tests are failing because we didn't update the assembly version for the il project. We need to also bump the version in the .il file for this particular project https://github.com/dotnet/corefx/blob/0e84b39fef10d6f220258b44a9698ccb2f364ee2/src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il#L40

**Comment by weshaggard** [comment]
No I don't believe this wouldn't require a modification of `InboxOn`, The issue is stemming from https://github.com/dotnet/corefx/commit/7ee51be7ed472a12266c8cc2ab25a9e1862c1d00#diff-c7bf0a95d7488e5968fd7aeb0078fda2 which explicitly pinned the version of the ref which I think is OK. The error is saying the ref is lower version then the implementation which should be valid. I'm not sure why it is complaining, do we need to suppress this check @ericstj?

**Comment by ericstj** [comment]
That's happening because the supported framework is applied to the cross-targeting-csproj reference, which will pull the max of all versions from that ref.  
https://github.com/dotnet/corefx/blob/0173b4186256de4d6445890b9e5a5ba6a22325c0/src/System.Security.Principal.Windows/pkg/System.Security.Principal.Windows.pkgproj#L5-L7
In this case the max will be the netfx version.  Which is why this is failing.

@weshaggard dealt with the same issue here: https://github.com/dotnet/corefx/commit/241403223aa5ffde3057c435070d2e0c88861657#diff-1345536edae72c012e218f0374fa418cR6

In his case he gets the support checks for the other frameworks from harvesting.  An alternate to this is to explicitly list SupportedFrameworks as follows:
https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/Documentation/coding-guidelines/package-projects.md#applicability-validation

**Comment by wtgodbe** [comment]
Test failure was a Jenkins issue

@dotnet-bot test Windows x86 Release Build

**Pull request merged**